src/transactions/TransactionFrame.cpp

```diff
@@ -740,6 +740,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                                   TransactionMeta& outerMeta)
 {
     ZoneScoped;
+    auto& internalErrorCounter = app.getMetrics().NewCounter(
+        {"ledger", "transaction", "internal-error"});
+
+    // We do not want to treat exception as internal-error for non-live ledger
+    // versions. The live ledger version is in the head of the ledger, and
+    // normally should be same as the app.getConfig().LEDGER_PROTOCOL_VERSION,
+    // which should also be same as maxAssertLedgerVersion here. However, right
+    // before a protocol upgrade, the LEDGER_PROTOCOL_VERSION (the software
+    // version) could lead the protocol version by 1, thus we need to offset it
+    // from the maxAssertLedgerVersion.
+    uint32_t maxAssertLedgerVersion =
```

Comment:
Nitpick: I think you mean `minAssertLedgerVersion`? Also "Assert" is a bit strange in this context since we actually never `assert` on internal error, but perhaps "Alert" instead? If you rename here, please update the comment to match.